### ISSUE TITLE
0.55.2 breaks collection typealias with compound existential type

### ISSUE THREAD

**Issue by jshier** [nicklockwood/SwiftFormat#1929]
As of 0.55.2, `public typealias Parameters = [String: any Any & Sendable]` is formatted into `public typealias Parameters = Sendable] & [String: any Any`, which is obviously broken. This also triggers a reindentation of the rest of the file, as the typealias is the first thing in the file.

**Comment by nicklockwood** [comment]
@jshier any idea which rule triggered this?

**Comment by jshier** [comment]
Sorry, my current config:
```
# file options

--symlinks ignore
--swiftversion 5.9

# format options

--commas inline
--comments indent
--decimalgrouping 3,5
--exponentcase lowercase
--exponentgrouping disabled
--extensionacl on-declarations
--fractiongrouping disabled
--ifdef no-indent
--importgrouping testable-top
--operatorfunc no-space
--nospaceoperators ..<, ...
--selfrequired validate
--someAny false
--stripunusedargs closure-only
--wraparguments preserve
--wrapcollections preserve
--wrapparameters preserve


# rules

--enable isEmpty
--disable andOperator
--disable opaqueGenericParameters
--disable wrapMultilineStatementBraces
```

**Comment by jshier** [comment]
For full context, you can run `swiftformat .` in [Alamofire](https://github.com/alamofire/alamofire).

**Comment by nicklockwood** [comment]
@jshier it's the `sortTypealiases` rule, so I suggest disabling that for now

**Comment by nicklockwood** [comment]
@calda would you mind taking a look at this? I've added a failing test on the develop branch

**Comment by nicklockwood** [comment]
@calda for now I've added a simple fix that just disables the rule for any types it might break. Ideally it should probably still be able to sort protocols with a type declaration, or ones that begin with `any` but I'll leave it up to you whether that is worth pursuing

**Comment by calda** [comment]
Sounds good! I'm traveling this week but will take a look when I'm home 😄

**Comment by nicklockwood** [comment]
@jshier fix landed in 0.55.3

**Comment by calda** [comment]
Here's a more robust fix that lets us support more complex cases like types with the `any` keyword: https://github.com/nicklockwood/SwiftFormat/pull/1938
